#! /bin/sh -e

# DP: ARM profiling fix

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 --fuzz 10 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 --fuzz 10 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- gcc/config/arm/linux-elf.h.old	Tue Jul 30 23:19:59 2002
+++ gcc/config/arm/linux-elf.h	Tue Jul 30 23:20:47 2002
@@ -304,3 +304,14 @@
 #undef  FP_DEFAULT
 #define FP_DEFAULT FP_SOFT3
 #endif
+
+/* Call the function profiler with a given profile label.  */
+#undef  FUNCTION_PROFILER
+#define FUNCTION_PROFILER(STREAM, LABELNO)  			\
+{									\
+  fprintf (STREAM, "\tbl\tmcount%s\n", NEED_PLT_GOT ? "(PLT)" : "");	\
+}
+
+#undef  CC1_SPEC
+#define CC1_SPEC "%{profile:-p}"
+
